compiler/lib/code.mli

```diff
@@ -181,9 +181,11 @@ type constant =
   | NativeString of Native_string.t
   | Float of float
   | Float_array of float array
+  | Int of int32
+  | Int32 of int32  (** Only produced when compiling to WebAssembly. *)
```

Comment:
I was expecting that Int32 and NativeInt would be used regardless of the backend and that generate would translate all to int.

Reply:
That is an option. One disadvantage of doing it that way is that then, functions like `int_bin_op` and `shift` in `Eval` have to deal with three types of integers instead of one.

On the other hand that is a rather small disadvantage. @vouillon Do you see other reasons not to produce `Int32` and `NativeInt` constructors when producing Javascript?

Reply:
I think it's a good thing to make it explicit that all these different integers share the same representation in JSOO.
In particular, it is clear that `int_bin_op` is correct and handles all the possible cases. Otherwise, since integer conversion functions are the identity, this function would have to deal with the three types of integers indeed, which could be somewhat confusing.

Reply:
I tend to be convinced by @vouillon's argument.